docs/eventing/channel-based-broker.md

```diff
@@ -0,0 +1,214 @@
+`Broker` into.
+
+Create the `ServiceAccount` objects.
+
+```shell
+kubectl -n default create serviceaccount eventing-broker-ingress
+kubectl -n default create serviceaccount eventing-broker-filter
+```
+
+Then give them the needed RBAC permissions:
+
+```shell
+kubectl -n default create rolebinding eventing-broker-ingress \
+  --clusterrole=eventing-broker-ingress \
+  --serviceaccount=default:eventing-broker-ingress
+kubectl -n default create rolebinding eventing-broker-filter \
+  --clusterrole=eventing-broker-filter \
+  --serviceaccount=default:eventing-broker-filter
+```
+
+Note that these commands each use three different objects, all named
+`eventing-broker-ingress` or `eventing-broker-filter`. The `ClusterRole` is
+installed with Knative Eventing
+[here](https://github.com/knative/eventing/blob/master/config/200-broker-clusterrole.yaml).
+The `ServiceAccount` was created two commands prior. The `RoleBinding` is
+created with this command.
+
+Create RBAC permissions granting access to shared configmaps for logging,
+tracing, and metrics configuration.
+
+_These commands assume the shared Knative Eventing components are installed in
+the `knative-eventing` namespace. If you installed the shared Knative Eventing
+components in a different namespace, replace `knative-eventing` with the name of
+that namespace._
+
+```shell
+kubectl -n knative-eventing create rolebinding eventing-config-reader-default-eventing-broker-ingress \
+  --clusterrole=eventing-config-reader \
+  --serviceaccount=default:eventing-broker-ingress
+kubectl -n knative-eventing create rolebinding eventing-config-reader-default-eventing-broker-filter \
+  --clusterrole=eventing-config-reader \
+  --serviceaccount=default:eventing-broker-filter
+```
+
+Now we can create the `Broker`. Note that this example uses the name `default`,
+but could be replaced by any other valid name. This example uses the defaults
+for Channel that we configured above in our `config-br-defaults` `ConfigMap`,
+and hence would use Kafka Channels.
+
+```shell
+cat << EOF | kubectl apply -f -
+apiVersion: eventing.knative.dev/v1beta1
+kind: Broker
+metadata:
+  namespace: default
+  name: default
+EOF
+```
+
+If you wanted to explicitly specify the Configuration of the Broker, you could do
+so by using spec.config, like so (manually overriding it to use InMemoryChannels).
+
+```shell
+cat << EOF | kubectl apply -f -
+apiVersion: eventing.knative.dev/v1beta1
+kind: Broker
+metadata:
+  namespace: default
+  name: broker-2
+spec:
+  config:
+    apiVersion: v1
+    kind: ConfigMap
+    name: imc-channel
+    namespace: knative-eventing
+EOF
+```
+
+Easiest way to create and control `Broker`s in your cluster is by using an automagic
+injection by labeling a namespace.
+
+## Handling failed event delivery
+
+Broker allows you to specify what to in the case if failed event delivery, say
+you have a consumer (function for example) that's failing to process the event.
+You can use `DeliverySpec` for it:
```

Comment:
It might make more sense to the user to call this `delivery` since `DeliverySpec` is only mentioned in the Go type. The example doesn't have `DeliverySpec` anywhere.